### PR TITLE
restore 404 handling, do not print stacktrace

### PR DESCRIPTION
**Pull request by danielnorberg**
Factored out status code checking into a separate reusable method called `requireSuccess`. 

This could perhaps do with some more refactoring. It seems a little odd that the `send()` method interprets a 404 as an error.

### CODE REVIEW

**Comment by danielnorberg** [comment]
@honnix PTAL

File: styx-cli/src/main/java/com/spotify/styx/cli/Main.java [hunk 1]
```diff
@@ -379,16 +379,21 @@ private void workflowShow() throws ExecutionException, InterruptedException, IOE
...
     CompletableFuture.allOf(workflowResponse, stateResponse).get();
 
-    final Workflow workflow = deserialize(workflowResponse.get(), Workflow.class);
-    final WorkflowState workflowState = deserialize(stateResponse.get(), WorkflowState.class);
+    final ByteString workflowPayload = requireSuccess(workflowResponse.get())
+        .payload().orElse(ByteString.EMPTY);
+    final ByteString statePayload = requireSuccess(stateResponse.get())
+        .payload().orElse(ByteString.EMPTY);
+
+    final Workflow workflow = deserialize(workflowPayload, Workflow.class);
```
Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: Do we get `null` if `workflowPayload` is `ByteString.EMPTY`?

File: styx-cli/src/main/java/com/spotify/styx/cli/Main.java [hunk 2]
```diff
@@ -484,22 +489,26 @@ private String apiUrl(CharSequence... parts) {
   }
 
   private ByteString send(Request request) throws ExecutionException, InterruptedException {
-    return sendAsync(request).toCompletableFuture().get();
+    // TODO: Move requireSuccess use into callers of send()
```
Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: A following PR to do for `TODO` I guess?

Comment (danielnorberg) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: The `send` method already threw `RuntimeException` on 404 so this just restores that behavior. I think that might not belong in `send` though hence the TODO.

https://github.com/spotify/styx/pull/130/files#diff-0a3c00857227c1cef7cc1e40e3d912acL454

Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: Yep, that was my point. I think that might not belong in `send`. Just asking your plan when to move it out.

File: styx-cli/src/main/java/com/spotify/styx/cli/Main.java (hunk 1 above)
Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: Although we shouldn't get it anyway...

Comment (danielnorberg) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: I guess the `AutoValue_Workflow` will throw NPE. That's true for other methods like `resourceShow` as well. The cli doesn't really seem to validate anything it gets from the styx api.

File: styx-cli/src/main/java/com/spotify/styx/cli/Main.java (hunk 2 above)
Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: cool.

**Comment by honnix** [comment]
👍

**Pull request merged**